src/schema/users.ts

```diff
@@ -1866,6 +1878,77 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         weekStart,
       };
     },
+    recoverStreak: async (
+      _,
+      __,
+      ctx: AuthContext,
+      info,
+    ): Promise<GQLUserStreak> => {
+      const { userId } = ctx;
+      const user = await ctx.con
+        .getRepository(User)
+        .findOneByOrFail({ id: userId });
+
+      const key = generateStorageKey(
+        StorageTopic.Streak,
+        StorageKey.Reset,
+        userId,
+      );
+      const oldStreakLength = Number(await getRedisObject(key));
+      const userDoesntHaveOldStreak =
+        !oldStreakLength || !isNumber(oldStreakLength);
+      if (userDoesntHaveOldStreak) {
+        throw new ValidationError('No streak to recover');
+      }
```

Comment:
Not 100% sure but isn't this 1-1 the same as for the query, maybe we can combine some logic in a fn?

Reply:
Updated! ✅